auth/src/android/common_android.cc

```diff
@@ -115,7 +115,10 @@ static const ErrorCodeMapping kFirebaseAuthCodes[] = {
     {"ERROR_UNSUPPORTED_FIRST_FACTOR", kAuthErrorUnsupportedFirstFactor},
     {"ERROR_EMAIL_CHANGE_NEEDS_VERIFICATION",
      kAuthErrorEmailChangeNeedsVerification},
-    {"ERROR_USER_CANCELLED", kAuthErrorMissingOrInvalidNonce},
```

Comment:
Also, super thanks for doing this work here.

Reply:
No problem!

So `kAuthErrorMissingOrInvalidNonce` is also already defined here: https://github.com/firebase/firebase-cpp-sdk/blob/6d5732ea7a428d43c2fae1e4ffe4a9eef8965326/auth/src/android/common_android.cc#L68

i.e. the `kAuthErrorMissingOrInvalidNonce` constant was defined twice in this file, and `kAuthErrorUserCancelled` was never defined.

Reply:
`kAuthErrorMissingOrInvalidNonce` is defined in `types.h` just above `kAuthErrorUserCancelled`.  It's then used to associate with an error details string in `common_android.cc`, but you're removing that string association.  `kAuthErrorUserCancelled` needs to be put back in `common_android.cc`, in the order that it was, but it needs to be associated with a different string as you've pointed out.  The previous string association of Nonce-failure is indeed incorrect.

Essentially `common_android.cc` contains a table of error codes mapped to detailed strings.  The error code `kAuthErrorUserCancelled` should still have a string still, just a different one, if that makes sense.

Reply:
Sorry, not sure I'm 100% understanding!

So `kAuthErrorMissingOrInvalidNonce` originally had two string associations within `common_android.cc` -- the one incorrectly mapped to `ERROR_USER_CANCELLED ` and the one here which is correctly mapped to `ERROR_MISSING_OR_INVALID_NONCE `: https://github.com/firebase/firebase-cpp-sdk/blob/6d5732ea7a428d43c2fae1e4ffe4a9eef8965326/auth/src/android/common_android.cc#L68

I removed one of the mappings (the incorrect one), and updated the `kAuthErrorUserCancelled` code to the correct string.

Reply:
Ah, I see your point that the Nonce error is associated with both the [FirebaseAuthInvalidCredentialsException](https://github.com/firebase/firebase-cpp-sdk/blob/0c8c8b29bc2d62d66c6ac49ff2c3fb04f815a687/auth/src/android/common_android.cc#L360) and the generic [FirebaseAuthException](https://github.com/firebase/firebase-cpp-sdk/blob/0c8c8b29bc2d62d66c6ac49ff2c3fb04f815a687/auth/src/android/common_android.cc#L378).

Are we 100% certain that the Firebase Android SDK cannot throw a FirebaseAuthException with an error code of ERROR_MISSING_OR_INVALID_NONCE, and that only the FirebaseAuthInvalidCredentialsException throws such error codes?

I ask because it makes me nervous to remove error codes from FirebaseAuthException if we correctly support them now (granted maybe with an incorrect string).

Reply:
👍